std/algorithm/searching.d

```diff
@@ -4192,9 +4192,26 @@ bool startsWith(alias pred = "a == b", R, E)(R doesThisStart, E withThis)
 if (isInputRange!R &&
     is(typeof(binaryFun!pred(doesThisStart.front, withThis)) : bool))
 {
-    return doesThisStart.empty
-        ? false
-        : binaryFun!pred(doesThisStart.front, withThis);
+    if (doesThisStart.empty)
+        return false;
+
+    // auto-decoding special case
+    static if (isNarrowString!R)
+    {
+        dchar front;
```

Comment:
Instead of having a temporary for `front`, how about declaring an alias for `binaryFun!pred` and replacing the assignments with `return`s?